src/servers/src/postgres/types.rs

```diff
@@ -58,6 +59,317 @@ pub(super) fn schema_to_pg(origin: &Schema, field_formats: &Format) -> Result<Ve
         .collect::<Result<Vec<FieldInfo>>>()
 }
 
+fn encode_array(
```

Comment:
Can we use some macro to simplify this?

Reply:
It's hard to find a common abstraction for these match branches. Using macro may fragment the code and sacrifice readability quite a bit.